clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -301,12 +303,15 @@ public void close(final Duration timeout) {
      * the network thread the time to close down cleanly.
      *
      * @param timeout Upper bound of time to wait for the network thread to close its resources
+     * @param membershipOperation the operation on consumer group membership that the consumer will perform when closing
      */
-    private void closeInternal(final Duration timeout) {
+    private void closeInternal(final Duration timeout, CloseOptions.GroupMembershipOperation membershipOperation) {
         long timeoutMs = timeout.toMillis();
         log.trace("Signaling the consumer network thread to close in {}ms", timeoutMs);
         running = false;
         closeTimeout = timeout;
+        requestManagers.consumerMembershipManager.ifPresent(membershipManager ->
```

Comment:
not saying this is wrong, just not very familiar with this part of the code -- why are we adding a leave group here that wasn't there before?

Reply:
Indeed, this is not elegant. I will see how to fix this one.